### ISSUE TITLE
Prefer book sources over Wikipedia for root training data

### ISSUE THREAD

**Issue by marctjones** [marctjones/klareco#422]
## Problem

Current corpus is 95.6% Wikipedia, 4.4% books. But quality analysis shows:
- Wikipedia: 93.9% parse rate, 50.3% perfect parses
- Books: 96-99% parse rate, 75-95% perfect parses
- Fundamento: 98.9% parse rate, 85.5% perfect parses

**Impact**: Training data is dominated by lower-quality source (Wikipedia).

## Current Extraction

`scripts/train_roots.sh` extracts Tier 1-3 sentences with no source preference. Since Wikipedia dominates the corpus, it dominates training data.

## Proposed Enhancement

Add source-based sampling to balance quality:

```python
# After tier and quality filtering
source = entry.get('source', 'unknown')

# Stratified sampling: prefer books over Wikipedia
import random
if source == 'wikipedia':
    # Keep 50% of Wikipedia sentences (reduce dominance)
    if random.random() > 0.5:
        continue
elif source in ['gutenberg', 'tekstaro']:
    # Keep ALL book sentences (they're rare and high quality)
    pass
```

**Result**: ~50/50 Wikipedia/Books balance in training data, dramatically improving average quality.

## Alternative: Source Weights

Instead of hard filtering, use source weights during training:

```python
# In train_root_embeddings.py
SOURCE_WEIGHTS = {
    'fundamento': 5.0,      # 5x weight
    'gutenberg': 2.0,       # 2x weight  
    'tekstaro': 2.0,        # 2x weight
    'wikipedia': 1.0,       # baseline
}

# During sampling
weight = SOURCE_WEIGHTS.get(source, 1.0)
# Use weight in loss computation or sampling probability
```

## Acceptance Criteria

- [ ] Book sentences are NOT undersampled relative to Wikipedia
- [ ] Training logs show source distribution of Ekzercaro data
- [ ] Validation set has similar source balance
- [ ] Documentation explains source preference rationale

## Expected Impact

- Higher average parse quality in training data
- Better root embeddings from cleaner co-occurrence signal
- More literary/formal language patterns (books) vs informal (Wikipedia)

## Files to Update

- `scripts/train_roots.sh` (inline extraction code)
- Or `scripts/train_root_embeddings.py` (if implementing source weights)

## Related

- Corpus analysis: Wikipedia 93.9%, Books 96-99% parse rates
- Fundamento is Tier 1 and highest quality (98.9%)
